pkg/skaffold/lint/lint.go

```diff
@@ -20,12 +20,61 @@ import (
 	"context"
 	"io"
 
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
+	schemautil "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/util"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 )
 
+var NewRunnerOpts = newRunnerOpts
+
+type RunnerOpts struct {
+	Context         context.Context
+	Out             io.Writer
+	Opts            config.SkaffoldOptions
+	CreateRunnerFxn func(ctx context.Context, out io.Writer, opts config.SkaffoldOptions) (runner.Runner, []schemautil.VersionedConfig, *runcontext.RunContext, error)
+}
+
+func newRunnerOpts(ctx context.Context, out io.Writer, opts config.SkaffoldOptions, createRunnerFxn func(ctx context.Context, out io.Writer, opts config.SkaffoldOptions) (runner.Runner, []schemautil.VersionedConfig, *runcontext.RunContext, error)) *RunnerOpts {
+	return &RunnerOpts{
+		Context:         ctx,
+		Out:             out,
+		Opts:            opts,
+		CreateRunnerFxn: createRunnerFxn,
+	}
+}
+
+// workaround to avoid acyclic dependencies and also to keep createRunner private for using createRunner in pkg/skaffold/lsp/handler.go
+func (ro *RunnerOpts) CreateRunner() (runner.Runner, []schemautil.VersionedConfig, *runcontext.RunContext, error) {
+	return ro.CreateRunnerFxn(ro.Context, ro.Out, ro.Opts)
+}
```

Comment:
This seems a bit hacky, is there any refactoring of packages we can do to avoid the cyclic dependency instead?

Reply:
I modified this to be be slightly less hacky and to have a similar concept in the lsp package directly.  The alternative here would be to move createRunner out of cmd and then export CreateRunner in that new package.  I wasn't sure of the best location, etc. though so I opted for a hack instead atm.